### PR TITLE
Add usage of CursorHoldI instead of CursorMovedI

### PR DESCRIPTION
**Pull request by nenkoru**
This PR features:
- Reduces amount of request that are being sent to an API. 
- A time after typing a symbol could be controlled and an API would not be bloated with outdated requests.

When I was using my own server I noticed that there are a lot of requests being made when I am still writing a code and not waiting for a suggestion to appear. I sincerely don't think that we need to generate a suggestion as soon as the user hits any symbol. And a request is still happening, server process it.
Vim basically waits certain amount of time until no new keys are pressed by the user and then fires up `CursorHoldI` event.

This autocmd is well documented and further info could be found [here](https://vimdoc.sourceforge.net/htmldoc/autocmd.html#CursorHold)
And time in ms that could be used to control the behaviour is `updatetime`.
By default it is 4000ms.
```vim
set updatetime=500
```

### CODE REVIEW

**Comment by nenkoru** [comment]
If an idea is supported - would also change README.md to setting of updatetime

**Comment by McPatate** [comment]
Hi @nenkoru, thanks for taking the time to submit your PRs !

I'm currently out of office, I'll take a look when I get back in a few days.

**Comment by McPatate** [comment]
Have you tried increasing the value of `opts.debounce_ms`? I'm a pretty fast typer so 80ms works for me.

Could you provide some data (if possible) with the number of calls that were emitted with the corresponding text that was typed?

File: lua/hfcc/completion.lua [hunk 1]
```diff
@@ -142,8 +143,9 @@ function M.create_autocmds()
   api.nvim_create_augroup(augroup, { clear = true })
 
   api.nvim_create_autocmd("InsertLeave", { pattern = "*", callback = M.cancel })
+  api.nvim_create_autocmd("CursorMovedI", { pattern = "*", callback = M.cancel })
 
-  api.nvim_create_autocmd("CursorMovedI", {
+  api.nvim_create_autocmd("CursorHoldI", {
```
Comment (wilfriedroset) [inline lua/hfcc/completion.lua:148]: could this be made configurable? this way user have the choice between tuning `opts.debounce_ms` or enabling `CursorHoldI`.

File: lua/hfcc/completion.lua [hunk 2]
```diff
@@ -125,6 +125,7 @@ function M.complete()
     api.nvim_win_set_cursor(0, { row_offset, col_offset })
 
     M.suggestion = nil
+    vim.defer_fn(M.schedule, 10)
```
Comment (wilfriedroset) [inline lua/hfcc/completion.lua:128]: noob question: could you clarify why is this needed?

Comment (nenkoru) [reply lua/hfcc/completion.lua:128]: To continue generating on <accept> of suggestion.
If to execute it right away it doesn't work, because of the way functions are executed.
It defers the execution after certain amount of time(could be wrong regarding time tho)

File: lua/hfcc/completion.lua (hunk 1 above)
Comment (nenkoru) [reply lua/hfcc/completion.lua:148]: For me debounce doesn't work, I tried increasing it but it didn't change a thing for me.
Didn't have time to screencast the behaviour.

Comment (McPatate) [reply lua/hfcc/completion.lua:148]: I'll try running some tests locally this week, keep you posted

Comment (McPatate) [reply lua/hfcc/completion.lua:148]: Found the culprit :)

cf #29

**Comment by McPatate** [comment]
I'll close this for now as I consider it fixed. Feel free to open an issue/PR if there is anything else you want to add that was discussed in this PR.
